resources/providers/azurelib/inventory/provider.go

```diff
@@ -157,6 +169,29 @@ func (p *Provider) ListAllAssetTypesByName(ctx context.Context, assetsGroup stri
 	return resAssets, nil
 }
 
+func (p *Provider) ListDiagnosticSettingsAssetTypes(ctx context.Context) ([]AzureAsset, error) {
+	p.log.Info("Listing Azure Diagnostic Monitor Settings")
+
+	var assets []AzureAsset
+
+	for subID, subName := range p.subscriptions {
+		pager := p.client.AssetDiagnosticSettingsPager(subID, nil)
+		res, err := readPager(ctx, pager)
+		if err != nil {
+			return nil, err
+		}
+		a, err := transformDiagnosticSettingsClientListResponses(res, subID, subName)
```

Comment:
Maybe a better name than `a`?

Reply:
I tend to follow [this](https://research.swtch.com/names) naming philosophy where the larger the scope of the variable, the more descriptive the name should be. 

So something that is used directly after initialization could be named single letter e.t.c.

But sure if we don't follow that rule, I could change it.